specs/core/1_custody-game.md

```diff
@@ -688,6 +692,18 @@ Append this to `process_final_updates(state)`:
 ```python
     # Clean up exposed RANDAO key reveals
     state.exposed_derived_secrets[current_epoch % EARLY_DERIVED_SECRET_PENALTY_MAX_FUTURE_EPOCHS] = []
+    # Reset withdrawable epochs if challenge records are empty
+    for index, validator in enumerate(state.validator_registry):
+        eligible = True
+        for records in (state.custody_chunk_challenge_records, state.bit_challenge_records):
+            for filter_func in (lambda rec: rec.challenger_index == index, lambda rec: rec.responder_index == index):
+                if len(list(filter(filter_func, records))) > 0:
+                    eligible = False
+        if eligible:
+            if validator.exit_epoch == FAR_FUTURE_EPOCH:
+                validator.withdrawable_epoch = FAR_FUTURE_EPOCH
+            else:
+                validator.withdrawable_epoch = validator.exit_epoch + MIN_VALIDATOR_WITHDRAWABILITY_DELAY
 ```
 
 In `process_penalties_and_exits`, change the definition of `eligible` to the following (note that it is not a pure function because `state` is declared in the surrounding scope):
```

Comment:
Both `process_penalties_and_exits` and `eligible` no longer exist in the phase 0 spec after #850.

Reply:
discuss in #1066